ping_test.go

```diff
@@ -264,3 +271,174 @@ func AssertFalse(t *testing.T, b bool) {
 		t.Errorf("Expected False, got True, Stack:\n%s", string(debug.Stack()))
 	}
 }
+
+func BenchmarkProcessPacket(b *testing.B) {
+	pinger, _ := NewPinger("127.0.0.1")
+
+	pinger.ipv4 = true
+	pinger.addr = "127.0.0.1"
+	pinger.network = "ip4:icmp"
+	pinger.id = 123
+	pinger.Tracker = 456
+
+	t := append(timeToBytes(time.Now()), intToBytes(pinger.Tracker)...)
+	if remainSize := pinger.Size - timeSliceLength - trackerLength; remainSize > 0 {
+		t = append(t, bytes.Repeat([]byte{1}, remainSize)...)
+	}
+
+	body := &icmp.Echo{
+		ID:   pinger.id,
+		Seq:  pinger.sequence,
+		Data: t,
+	}
+
+	msg := &icmp.Message{
+		Type: ipv4.ICMPTypeEchoReply,
+		Code: 0,
+		Body: body,
+	}
+
+	msgBytes, _ := msg.Marshal(nil)
+
+	pkt := packet{
+		nbytes: len(msgBytes),
+		bytes:  msgBytes,
+		ttl:    24,
+	}
+
+	for k := 0; k < b.N; k++ {
+		pinger.processPacket(&pkt)
+	}
+}
+
+func BenchmarkProcessPacketOld(b *testing.B) {
+	pinger, _ := NewPinger("127.0.0.1")
+
+	pinger.ipv4 = true
+	pinger.addr = "127.0.0.1"
+	pinger.network = "ip4:icmp"
+	pinger.id = 123
+	pinger.Tracker = 456
+
+	type IcmpData struct {
+		Bytes   []byte
+		Tracker int64
+	}
+
+	byteSliceOfSize := func(n int) []byte {
+		b := make([]byte, n)
+		for i := 0; i < len(b); i++ {
+			b[i] = 1
+		}
+
+		return b
+	}
+
+	t := timeToBytes(time.Now())
+	if pinger.Size-timeSliceLength != 0 {
+		t = append(t, byteSliceOfSize(pinger.Size-timeSliceLength)...)
+	}
+
+	data, _ := json.Marshal(IcmpData{Bytes: t, Tracker: pinger.Tracker})
+
+	body := &icmp.Echo{
+		ID:   pinger.id,
+		Seq:  pinger.sequence,
+		Data: data,
+	}
+
+	msg := &icmp.Message{
+		Type: ipv4.ICMPTypeEchoReply,
+		Code: 0,
+		Body: body,
+	}
+
+	msgBytes, _ := msg.Marshal(nil)
+
+	pkt := packet{
+		nbytes: len(msgBytes),
+		bytes:  msgBytes,
+	}
+
+	processPacket := func(p *Pinger, recv *packet) error {
+		var bytes []byte
+		var proto int
+		if p.ipv4 {
+			if p.network == "ip" {
+				bytes = ipv4Payload(recv.bytes)
```

Comment:
This should be just `recv` now, I will add unit testing and `go vet` to the circleci config to catch this in the future